contracts/modules/LoanManager.sol

```diff
@@ -0,0 +1,256 @@
+        emit CollateralRemoved(address(_wallet), _provider, _loanId, _collateral, _collateralAmount);
+    }
+
+    /**
+     * @dev Increases the debt by borrowing more token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _provider The address of the provider to use.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _debtToken The token borrowed.
+     * @param _debtAmount The amount of token to borrow.
+     */
+    function addDebt(
+        BaseWallet _wallet, 
+        address _provider, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount
+    ) 
+        external
+        onlyWhenUnlocked(_wallet) 
+    {
+        bytes memory methodData = abi.encodeWithSignature(
+            "addDebt(address,bytes32,address,uint256,address[])", 
+            address(_wallet), 
+            _loanId,
+            _debtToken,
+            _debtAmount,
+            providers[_provider].oracles
+            );
+        (bool success, ) = delegateToProvider(_provider, methodData);
+        require(success, "LoanManager: request to provider failed");
+        emit DebtAdded(address(_wallet), _provider, _loanId, _debtToken, _debtAmount);
+    }
+
+    /**
+     * @dev Decreases the debt by repaying some token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _provider The address of the provider to use.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _debtToken The token to repay.
+     * @param _debtAmount The amount of token to repay.
+     */
+    function removeDebt(
+        BaseWallet _wallet, 
+        address _provider, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount 
+    ) 
+        external
+        onlyWhenUnlocked(_wallet)
+    {
+        bytes memory methodData = abi.encodeWithSignature(
+            "removeDebt(address,bytes32,address,uint256,address[])", 
+            address(_wallet), 
+            _loanId,
+            _debtToken,
+            _debtAmount,
+            providers[_provider].oracles
+            );
+        (bool success, ) = delegateToProvider(_provider, methodData);
+        require(success, "LoanManager: request to provider failed");
+        emit DebtRemoved(address(_wallet), _provider, _loanId, _debtToken, _debtAmount);
+    }
+
+    /**
+     * @dev Gets information about a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _provider The address of the provider to use.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @return a status [0: no loan, 1: loan is safe, 2: loan is unsafe and can be liquidated, 3: unable to provide info] and the estimated ETH value of the loan
+     * combining all collaterals and all debts. When status = 1 it represents the value that could still be borrowed, while with status = 2
```

Comment:
"and the estimated ETH value of the loan combining all collaterals and all debts" -> "and a value (in ETH) representing the value that could still be borrowed when status = 1; or the value of the collateral that should be added to avoid liquidation when status = 2"

Reply:
Done.